version: "3.9"

services:
  user1:
    build:
      context: ..
      dockerfile: docker/Dockerfile
    container_name: fachat_user1
    hostname: user1
    environment:
      - USER_NAME=Alice
      - PORT=5000
      - MY_MAC=02:42:ac:1c:00:02
      - SECRET_KEY=c8a6d95a3b912f58e9b2f214f1e3e13f627c04f7a1cb0d8c08d27e7b45e76e90
      - SESSION_COOKIE_NAME=session_app1
    ports:
      - "5005:5000"  # Para acceder desde host
    networks:
      fachat_net:
        ipv4_address: 172.28.0.2
    command: ["python3", "web/app.py"]

  user2:
    build:
      context: ..
      dockerfile: docker/Dockerfile
    container_name: fachat_user2
    hostname: user2
    environment:
      - USER_NAME=Bob
      - PORT=5000
      - MY_MAC=02:42:ac:1c:00:03
      - SECRET_KEY=6f88f120cecd5a6eeb1a2d7b92fd1b1434cb4c692a933b2f3ec90bb80e7546e2
      - SESSION_COOKIE_NAME=session_app2
    ports:
      - "5006:5000"  # Mapeo distinto al host
    networks:
      fachat_net:
        ipv4_address: 172.28.0.3
    command: ["python3", "web/app.py"]

networks:
  fachat_net:
    driver: bridge
    ipam:
      config:
        - subnet: 172.28.0.0/16
